test/test.js

```diff
@@ -24,21 +15,57 @@ jest.mock('@octokit/rest', () => {
 
 describe('GitHubActions', () => {
   let githubActions;
+  let mockListRepoWorkflows;
 
   beforeEach(() => {
     githubActions = new GitHubActions();
+    mockListRepoWorkflows = githubActions.octokit.rest.actions.listRepoWorkflows;
   });
 
   it('should create an instance of GitHubActions', () => {
     expect(githubActions).toBeInstanceOf(GitHubActions);
   });
 
-  it('should return correct colors for GitHub Actions states', async () => {
+  it('should return the correct color for a pending action', async () => {
+    githubActions.octokit.rest.actions.listRepoWorkflows.mockResolvedValueOnce({
+      data: {
+        workflows: [{ id: 1, state: 'active' }],
+      },
+    });
+
+    const actions = await githubActions.getGitHubActions();
+    const status = githubActions.getStatus(actions);
+    const color = githubActions.getColor(status);
+
+    expect(color).toBe('#FFA500');
+  });
+
+  it('should return the correct color for a successful action', async () => {
+    githubActions.octokit.rest.actions.listRepoWorkflows.mockResolvedValueOnce({
+      data: {
+        workflows: [{ id: 1, state: 'completed' }],
+      },
+    });
+
     const actions = await githubActions.getGitHubActions();
-    const points = githubActions.generatePoints(actions);
+    const status = githubActions.getStatus(actions);
+    const color = githubActions.getColor(status);
 
-    expect(points[0].color).toBe('#FFA500'); // pending
-    expect(points[1].color).toBe('#00FF00'); // success
-    expect(points[2].color).toBe('#FF0000'); // failure
+    expect(color).toBe('#00FF00');
   });
+
+  it('should return the correct color for a failed action', async () => {
+    githubActions.octokit.rest.actions.listRepoWorkflows.mockResolvedValueOnce({
+      data: {
+        workflows: [{ id: 1, state: 'failed' }],
+      },
+    });
+
+    const actions = await githubActions.getGitHubActions();
+    const status = githubActions.getStatus(actions);
+    const color = githubActions.getColor(status);
+
+    expect(color).toBe('#FF0000');
+  });
+
```

Comment:
You need a test that handles mixed workflows. 

E.g.: 
- 2 failed 1 success
- 3 failed
- 1 success 1 pending 1 failed
...